src/lightning_lite/cli.py

```diff
@@ -0,0 +1,151 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from argparse import ArgumentParser, Namespace
+from typing import List, Tuple
+
+import torch.distributed.run as torchrun
+
+from lightning_lite.accelerators import CPUAccelerator, CUDAAccelerator, MPSAccelerator
+
+_SUPPORTED_ACCELERATORS = ("cpu", "gpu", "cuda", "mps", "tpu")
+_SUPPORTED_STRATEGIES = (None, "ddp", "dp", "deepspeed")
+_SUPPORTED_PRECISION = ("64", "32", "16", "bf16")
+
+
+def _parse_args() -> Tuple[Namespace, List[str]]:
+    parser = ArgumentParser(description="Launch your script with the Lightning Lite CLI.")
+    parser.add_argument("script", type=str, help="Path to the Python script with Lightning Lite inside.")
+    parser.add_argument(
+        "--accelerator",
+        type=str,
+        default="cpu",
+        choices=_SUPPORTED_ACCELERATORS,
+        help="The hardware accelerator to run on.",
+    )
+    parser.add_argument(
+        "--strategy",
+        type=str,
+        default=None,
+        choices=_SUPPORTED_STRATEGIES,
+        help="Strategy for how to run across multiple devices.",
+    )
+    parser.add_argument(
+        "--devices",
+        type=str,
+        default="1",
+        help=(
+            "Number of devices to run on (``int``), which devices to run on (``list`` or ``str``), or ``'auto'``."
+            " The value applies per node."
+        ),
+    )
+    parser.add_argument(
+        "--num-nodes",
```

Comment:
nit: I would prefer the `--num_nodes` pattern (underscore instead of dash), as it matches arguments in Python.

Reply:
Good point. Actually, we should probably expect the _ vs. - to be a common typo. The issue is, if we only support the num_nodes (underscore) version, and user types num-nodes (hypen) by accident, the argument does not get parsed by the cli and passed down as script args. This silent behavior, in general, is worrying me a bit tbh.